src/create-reminder-signup/lambda/lambda.ts

```diff
@@ -1,65 +1,79 @@
 import { APIGatewayProxyResult } from 'aws-lambda';
 import * as AWS from 'aws-sdk';
 import * as SSM from 'aws-sdk/clients/ssm';
-import { getParamFromSSM, ssmStage } from '../../lib/ssm';
-import { getOrCreateIdentityIdByEmail, IdentityResult } from '../lib/identity';
-import {APIGatewayEvent, oneOffSignupValidator} from './models';
+import { Pool } from 'pg';
+import { createDatabaseConnectionPool } from '../../lib/db';
+import {
+	getDatabaseParamsFromSSM,
+	getParamFromSSM,
+	ssmStage,
+} from '../../lib/ssm';
+import { writeOneOffSignup } from '../lib/db';
+import { getOrCreateIdentityIdByEmail } from '../lib/identity';
+import {
+	APIGatewayEvent,
+	oneOffSignupFromRequest,
+	oneOffSignupValidator,
+	ValidationErrors,
+} from './models';
 
 const ssm: SSM = new AWS.SSM({ region: 'eu-west-1' });
 
+const dbConnectionPoolPromise: Promise<Pool> = getDatabaseParamsFromSSM(
+	ssm,
+).then(createDatabaseConnectionPool);
 const identityAccessTokenPromise: Promise<string> = getParamFromSSM(
 	ssm,
 	`/support-reminders/idapi/${ssmStage}/accessToken`,
 );
 
-export const handler = (
+export const handler = async (
```

Comment:
I remember there was a team which had an issue with async handlers for node.js lambdas...

Reply:
haha yes, that was us. Curiously the aws advice was not to do it until you hit a problem, but we don't want to lose data so we should really just do it now.
I had to do it for referrals here: https://github.com/guardian/contributions-referrals/pull/26

Reply:
i'll do this in a follow up PR - I think it's potentially an issue regardless of the `async` keyword